R/run_mcmc.R

```diff
@@ -110,14 +110,37 @@ run_mcmc_p <- function(dt, priorObj, n.chains, n.adapt, n.burn, n.iter, seed, pa
     cat("seed_list:", seed_list, "number of dataset", length(dt), "\n")
   )
 
-  message(n.cores, " clusters are being used")
+  ps_message(n.cores, " clusters are being used")
   cl <- parallel::makeCluster(n.cores)
   doParallel::registerDoParallel(cl)
 
+  # Load psborrow on the clusters
+  ## Function to see whether psborrow is being used in a development context
+  is_psborrow_dev <- function() {
+    desc <- file.path("DESCRIPTION")
+    if (file.exists(desc)) {
+      content <- read.dcf(desc)
+      pkg <- content[1, ][["Package"]]
+      return(pkg == "psborrow")
+    }
+    return(FALSE)
+  }
+  ## Flag for global options to pass to clusters
+  local.psborrow.quiet <- getOption('psborrow.quiet')
+  parallel::clusterExport(cl, varlist =  c('local.psborrow.quiet','is_psborrow_dev'), envir = environment())
+  parallel::clusterEvalQ(cl, {
+    try(
+      if(is_psborrow_dev()) {
+        pkgload::load_all()
+      }
+    )
+    library(psborrow)
```

Comment:
Do we not want this in an "else" block, otherwise we are loading the package twice in development and risk conflicts between the development version and the installed version